### ISSUE TITLE
Fails to build with Go 1.5.1 and can't download 1.4.2

### ISSUE THREAD

**Issue by inkel** [prometheus/alertmanager#129]
Hi everyone. I'm trying to build the project on my local environment, but it's failing because I don't have Go 1.4.2 installed, and it's also failing to download it because the download URL is returning 404:

```
inkel@miralejos ~/dev/prometheus/alertmanager (master=)
$ make test
Go version 1.4.2 required but not found in PATH.
About to download and install go1.4.2 to /Users/inkel/dev/prometheus/alertmanager/.build/go1.4.2
Abort now if you want to manually install it system-wide instead.

mkdir -p /Users/inkel/dev/prometheus/alertmanager/.build/go1.4.2
curl -L https://golang.org/dl/go1.4.2.darwin-amd64-osx10.9.5.tar.gz | tar -C /Users/inkel/dev/prometheus/alertmanager/.build/go1.4.2 --strip 1 -xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    98  100    98    0     0     88      0  0:00:01  0:00:01 --:--:--    89
100   127  100   127    0     0     57      0  0:00:02  0:00:02 --:--:--     0
tar: Unrecognized archive format
tar: Error exit delayed from previous errors.
make: *** [/Users/inkel/dev/prometheus/alertmanager/.build/go1.4.2/bin/go] Error 1

inkel@miralejos ~/dev/prometheus/alertmanager (master=)
$ wget https://golang.org/dl/go1.4.2.darwin-amd64-osx10.9.5.tar.gz
--2015-10-06 20:15:14--  https://golang.org/dl/go1.4.2.darwin-amd64-osx10.9.5.tar.gz
Resolving golang.org... 64.233.186.141, 2800:3f0:4003:c00::8d
Connecting to golang.org|64.233.186.141|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://storage.googleapis.com/golang/go1.4.2.darwin-amd64-osx10.9.5.tar.gz [following]
--2015-10-06 20:15:14--  https://storage.googleapis.com/golang/go1.4.2.darwin-amd64-osx10.9.5.tar.gz
Resolving storage.googleapis.com... 64.233.186.128, 2800:3f0:4003:c00::80
Connecting to storage.googleapis.com|64.233.186.128|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2015-10-06 20:15:15 ERROR 404: Not Found.
```

This is the version I'm running locally:

```
inkel@miralejos ~/dev/prometheus/alertmanager (master=)
$ go version
go version go1.5.1 darwin/amd64
```

Note that assigning `GO_VERSION` environment variable does the trick:

```
$ GO_VERSION=1.5.1 make test
mkdir -p /Users/inkel/dev/prometheus/alertmanager/.build/gopath/src/github.com/prometheus/
ln -s /Users/inkel/dev/prometheus/alertmanager /Users/inkel/dev/prometheus/alertmanager/.build/gopath/src/github.com/prometheus/alertmanager
GOPATH=/Users/inkel/dev/prometheus/alertmanager/.build/gopath /usr/local/bin/go get -d
touch dependencies-stamp
GOPATH=/Users/inkel/dev/prometheus/alertmanager/.build/gopath /usr/local/bin/go test ./...
?   	_/Users/inkel/dev/prometheus/alertmanager	[no test files]
ok  	_/Users/inkel/dev/prometheus/alertmanager/config	0.019s
?   	_/Users/inkel/dev/prometheus/alertmanager/config/generated	[no test files]
ok  	_/Users/inkel/dev/prometheus/alertmanager/manager	0.023s
?   	_/Users/inkel/dev/prometheus/alertmanager/web	[no test files]
?   	_/Users/inkel/dev/prometheus/alertmanager/web/api	[no test files]
?   	_/Users/inkel/dev/prometheus/alertmanager/web/blob	[no test files]
```

**Comment by grobie** [comment]
The problem is the OSX specific `RELEASE_SUFFIX` we set in [Makefile.COMMON](https://github.com/prometheus/alertmanager/blob/master/Makefile.COMMON#L48). The implementation assumes that a go version exists for every return value of `sw_vers -productVersion`. The only available suffix is `10.8` though.

As >go1.5 does not longer use such release suffixes in OSX tarball locations, we should just update the Makefile.COMMON from prometheus/utils. Though, there is another osx specific problem in their we need to fix first.

**Comment by fabxc** [comment]
This issue does not affect `master`.
